src/sentry/models/rule.py

```diff
@@ -58,6 +58,15 @@ def get_for_project(cls, project_id):
             cache.set(cache_key, rules_list, 60)
         return rules_list
 
+    @property
+    def created_by(self):
+        created_activity = RuleActivity.objects.filter(
+            rule=self, type=RuleActivityType.CREATED.value
+        ).first()
```

Comment:
We generally try to avoid `first` because it forces a sort order on the results that you might not expect. If you only ever expect there to be one row, then `get` is fine. If there might be more, sort  by `id` and then `.first` should be fine. We might need to change this anyway since we'll need to fetch in bulk in the serializer.

Reply:
I agree about fetching in bulk for the serializer - but I think I will leave this in so it could be used as a sort field.

Reply:
If we sort on this then it'll have a similar problem - we need to fetch each row before we can sort. I think we'll need a way to prefetch this for a list of objects so that we only do this logic if we haven't already fetched the cached value.

Reply:
Aah I see. I can change it so that we can somehow prefetch these in bulk for sorting as part of another PR (like you suggested in another comment). We don't currently have a direct use-case, but I'm thinking of leaving this function in (changing it's implementation away from .first and towards .get in a try/except) for the future if we ever need created_by for a single rule.

Reply:
@wedamija i wonder if we can add a lint rule or something to complain and stop using .first().